src/data/zmod/quadratic_reciprocity.lean

```diff
@@ -41,6 +41,30 @@ lemma euler_criterion {a : zmodp p hp} (ha : a ≠ 0) :
 λ h, let ⟨y, hy⟩ := (euler_criterion_units hp).2 (show units.mk0 _ ha ^ (p / 2) = 1, by simpa [units.ext_iff]) in
   ⟨y, by simpa [units.ext_iff] using hy⟩⟩
 
+lemma neg_one_is_square_iff_mod_four_ne_three :
```

Comment:
It's not likely that people will be searching for this with tab completion often, but, I wonder, should we have better conventions for naming theorems with `square` and `pow_two`? For example, this could be `pow_two_eq_neg_one_iff_mod_four_ne_three`.